main/src/ca/uwaterloo/flix/api/lsp/provider/completion/BuiltinTypeCompleter.scala

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 Paul Butcher
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package ca.uwaterloo.flix.api.lsp.provider.completion
+
+import ca.uwaterloo.flix.api.lsp.{CompletionItem, Index, InsertTextFormat, TextEdit}
+import ca.uwaterloo.flix.api.lsp.provider.CompletionProvider.Priority
+import ca.uwaterloo.flix.language.ast.TypedAst
+
+object BuiltinTypeCompleter {
+
+  def getBuiltinTypeCompletions(priority: String => String)(implicit context: CompletionContext, index: Index, root: TypedAst.Root): Iterable[CompletionItem] = {
```

Comment:
I think we might aswell start by changing these functions to return `Iterable[Completion]` and then calling `toCompletionItem` in `Completer`.